src/wallet/bitcoin.ts

```diff
@@ -169,27 +172,28 @@ export class InMemoryBitcoinWallet implements BitcoinWallet {
   }
 
   public async close(): Promise<void> {
-    const tasks = [];
-    if (this.pool.opened) {
-      tasks.push(this.pool.close());
-    }
-
-    if (this.chain.opened) {
-      tasks.push(this.chain.close());
-    }
-
-    if (this.walletdb.opened) {
-      tasks.push(this.walletdb.close());
-    }
-
-    await Promise.all(tasks);
+    await axios.request({
+      ...this.rpcRequestConfig,
+      data: {
+        jsonrpc: "1.0",
+        method: "unloadwallet",
+        params: []
+      }
+    });
   }
 
-  private assertNetwork(network: string): void {
-    if (network !== this.network.type) {
-      throw new Error(
-        `This wallet is only connected to the ${this.network.type} network and cannot perform actions on the ${network} network`
+  private async assertNetwork(network: Network): Promise<void> {
+    const res = await axios.request({
+      ...this.rpcRequestConfig,
+      data: { jsonrpc: "1.0", method: "getblockchaininfo", params: [] }
+    });
+
+    if (res.data.result.chain !== network) {
+      return Promise.reject(
+        `This wallet is only connected to the ${network} network and cannot perform actions on the ${network} network`
       );
     }
   }
 }
+
+export type Network = "bitcoin" | "testnet" | "regtest";
```

Comment:
I see this already making problems :D 
See this bug: https://github.com/comit-network/comit-rs/issues/2438

I'm not sure how we want to solve it in the long run, but since you are using `getblockchaininfo` the accepted values are: 

```suggestion
export type Network = "main" | "test" | "regtest";
```

Reply:
Oh, no! Never mind then...